src/test/java/com/flightstats/hub/dao/aws/S3WriteQueueTest.java

```diff
@@ -0,0 +1,118 @@
+package com.flightstats.hub.dao.aws;
+
+import com.flightstats.hub.dao.aws.writeQueue.WriteQueueConfig;
+import com.flightstats.hub.metrics.StatsdReporter;
+import com.flightstats.hub.model.ChannelContentKey;
+import com.flightstats.hub.model.Content;
+import com.flightstats.hub.model.ContentKey;
+import com.flightstats.hub.spoke.SpokeWriteContentDao;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+@Slf4j
+public class S3WriteQueueTest {
+
+    private @Mock SpokeWriteContentDao spokeWriteContentDao;
+    private @Mock S3SingleContentDao s3SingleContentDao;
+    private @Mock StatsdReporter statsdReporter;
+    private @Mock ChannelContentKey key;
+    private @Mock ContentKey contentKey;
+    private WriteQueueConfig writeQueueConfig = WriteQueueConfig.builder()
+            .queueSize(20)
+            .threads(2)
+            .build();
+    private final static String CHANNEL_NAME = "testy_test";
+    private final static long AGE_MILLIS = 666;
+    private S3WriteQueue s3WriteQueue;
+
+    @Before
+    public void setup() {
+        initMocks(this);
+        s3WriteQueue = new S3WriteQueue(
+                spokeWriteContentDao,
+                s3SingleContentDao,
+                statsdReporter,
+                writeQueueConfig);
+    }
+
+    @Test
```

Comment:
can we start using junit 5 for new tests/ updates?

Reply:
junit 5 is not currently installed in the repo, build.gradle: `testCompile 'junit:junit:4.+'` 
I still have a branch to introduce it, but other things have been prioritized.